### ISSUE TITLE
Update to 8.4.6 adds a thousand buttons to the toolbar

### ISSUE THREAD

**Issue by ygoe** [notepad-plus-plus/notepad-plus-plus#12403]
### Description of the Issue
I've just updated NPP to version 8.4.6 as suggested by the application itself and now the toolbar overflows at any window width. There's a very big number of buttons in there now, most of them live in the overflow menu. This doesn't make the least of sense. Where do all these new button come from and how do I get rid of them?

### Steps to Reproduce the Issue
1. Update to version 8.4.6
2. Look at the toolbar

### Expected Behavior
No change to previous version

### Actual Behavior
Loads of stuff that doesn't even fit in there, no idea what it all is

### Debug Information
Notepad++ v8.4.6   (32-bit)
Build time : Sep 25 2022 - 19:55:26
Path : C:\Program Files (x86)\Notepad++\notepad++.exe
Command Line : "C:\Program Files (x86)\Notepad++\change.log" 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 21H2
OS Build : 19044.2006
Current ANSI codepage : 1252
Plugins : 
    CustomizeToolbar (4.2)
    JSMinNPP (1.2006)
    mimeTools (2.8)
    NppConverter (4.4)
    NppExport (0.4)
    NppSaveAsAdmin (1.0.211)
    XMLTools (3.1.1.6)

Here's a screenshot of the window in full-screen after clicking on the overflow button at the end of the toolbar:

![grafik](https://user-images.githubusercontent.com/868315/198259675-2309dbb6-2b81-4d49-b335-8600a8909815.png)

Normally the toolbar should only be half as long.

**Comment by ygoe** [comment]
Oh, and I added a custom button with the custom button plugin, which is now gone. Maybe it's at the end and never visible.

**Comment by alankilborn** [comment]
Please remove the CustomizeToolbar plugin and report results after restarting Notepad++.  Also post updated Debug Info so that we can confirm CustomizeToolbar has really been removed.

**Comment by ygoe** [comment]
Here's the debug info:

Notepad++ v8.4.6   (32-bit)
Build time : Sep 25 2022 - 19:55:26
Path : C:\Program Files (x86)\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 21H2
OS Build : 19044.2006
Current ANSI codepage : 1252
Plugins : 
    JSMinNPP (1.2006)
    mimeTools (2.8)
    NppConverter (4.4)
    NppExport (0.4)
    NppSaveAsAdmin (1.0.211)
    XMLTools (3.1.1.6)

And the screenshot, with still more buttons than fit in my regular window size:

![grafik](https://user-images.githubusercontent.com/868315/198272663-ab2da0c0-ccc9-452d-9f78-750f87138d46.png)

For comparison, this is how it looks on another computer with 8.4.5 and the same initial configuration (with CustomToolbar that should only add a single button and move it in the middle):

![grafik](https://user-images.githubusercontent.com/868315/198272880-cd677e39-a630-44ac-a2ae-e66fe0739447.png)

**Comment by alankilborn** [comment]
Interesting.  Can you state what some of the extra buttons do, or maybe what their hover text says, if anything?  

For example, these are some that are obviously specific to your strange situation, and it might be key to knowing how to get rid of them if we know what they do:

![image](https://user-images.githubusercontent.com/77065706/198277430-e91a6181-6866-4b81-ba56-0cea212a3bbd.png)

**Comment by ygoe** [comment]
From the black J button on:
```
JSON Viewer
Check XML syntax now
Validate now
First error (this is the first red icon you showed)
Previous error
Next error
Last error
Pretty print
Pretty print - indent attributes
Pretty print - indent only
Linearize
Current XML Path
Current XML Path with predicates
Evaluate XPath expression...
XSL Transformation...
Escape characters in selection (<> → lt;gt;)
Unescape characters in selection (lt;gt; → <>)
Comment selection
Uncomment selection
Options...
```

Looks like they come from the XML plugin. Something in the update must have triggered it to forget all its settings and clutter the toolbar with all the buttons it has.

I installed the Custom Toolbar plugin again and noticed it has now a newer version. Maybe the old one was incompatible with the new NPP application? I don't know what's happening there behind the scenes of plugins. And again, all buttons were added. I looked in the toolbar customisation dialog and saw that all available buttons were added, there was nothing more to add. After resetting the toolbar it looks more like before, but still with all the XML buttons. They disappear after fixing the XML plugin configuration (and a restart).

I've had trouble with the NPP toolbar before, mostly with my single added custom button and plugin buttons AFAIR. The whole system doesn't look very robust to me. I know such experience from using locked-down binary serialisation of internal data structures for configuration instead of employing a proper defined interface format. Internal data structures change often and leave user configuration corrupted and have unpredictable effects. Maybe that's the case here, too?

**Comment by alankilborn** [comment]
>I've had trouble with the NPP toolbar before

I've had some small amount of trouble with it myself when using the _CustomizeToolbar_ plugin.  I've never seen a problem with the native toolbar buttons.

Likely the culprit for your current issue is the plugin.

**Comment by ygoe** [comment]
Wow, this happens on another computer, after first upgrading the Custom Toolbar plugin and then NPP:

![grafik](https://user-images.githubusercontent.com/868315/198401751-b60a89ed-7a3d-46f6-8e00-33a75bc70e41.png)

A complete mess. Will it be like that after every upgrade now? I may consider stopping upgrading anything of that.

BTW: The upgrades view of the plugin manager seems broken. Custom Toolbar was installed 4.2, available 5.3. No upgrades offered for anything in the middle tab of that window.

**Comment by ArkadiuszMichalski** [comment]
@ygoe The best would be to remove old `Customize Toolbar` plugin folder and leave only `_CustomizeToolbar` (which is the default in current versions of this plugin). Then, in the plug-in settings, select the buttons you want to be visible again. Or remove all `CustomizeToolbar` folders (manually, directly on your `plugins` folder) and install latest version of this plugin.

You can also remove old `Npp\plugins\Config\CustomizeToolbar.dat` file. Also if you have own rules in `Npp\plugins\Config\CustomizeToolbar.btn` check if your definitions are consistent with `Customize Toolbar 5.2` version.

Your [first problem](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/12403#issue-1425396836) were probably because you upgraded Notepad++ and the new version added some new built-in buttons. The `Customize Toolbar` plugin is not prepared for it and sometimes it can lose its settings (then it shows more or all the buttons it recognizes). But that's a plugin problem and should be reported to the author of this plugin.
